src/Zicht/Bundle/SolrBundle/Manager/SolrManager.php

```diff
@@ -131,6 +131,41 @@ public function updateBatch($records, $incrementCallback = null, $errorCallback
         return [$n, $i];
     }
 
+    /**
+     * Extracts as batch.
+     *
+     * @param array $records
+     * @param callable|null $incrementCallback
+     * @param callable|null $errorCallback
+     *
+     * @return array
+     */
+    public function extractBatch($records, $incrementCallback = null, $errorCallback = null)
+    {
+        $n = $i = 0;
+        foreach ($records as $record) {
+            if ($mapper = $this->getMapper($record)) {
```

Comment:
Taste/opinion: I don't think assignment in an `if` statement is a good idea. Better to check on `null` and break out of the loop with `continue` so the code below is outside the `if`.